src/Plugin/QueueWorker/MpxImporterQueueWorker.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+namespace Drupal\media_mpx\Plugin\QueueWorker;
+
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\Core\Queue\QueueWorkerBase;
+use Drupal\media_mpx\DataObjectImporter;
+use Drupal\media_mpx\MpxImportTask;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\media_mpx\DataObjectFactoryCreator;
+
+/**
+ * Process mpx imports.
+ *
+ * This queue has been made to mitigate the process intensive nature
+ * of importing mpx objects.
+ *
+ * @QueueWorker(
+ *   id="media_mpx_importer",
+ *   title="mpx import queue worker",
+ *   cron={
+ *     "time"=15
+ *   }
+ * )
+ */
+class MpxImporterQueueWorker extends QueueWorkerBase implements ContainerFactoryPluginInterface {
+
+  /**
+   * The factory used to load a complete mpx object.
+   *
+   * @var \Drupal\media_mpx\DataObjectFactoryCreator
+   */
+  protected $dataObjectFactoryCreator;
+
+  /**
+   * A class to import mpx objects.
+   *
+   * @var \Drupal\media_mpx\DataObjectImporter
+   */
+  private $dataObjectImporter;
+
+  /**
+   * The entity type manager used to load entities.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  private $entityTypeManager;
+
+  /**
+   * MpxImporterQueueWorker constructor.
+   *
+   * @param array $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin_id for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entityTypeManager
+   *   An entity type manager.
+   * @param \Drupal\media_mpx\DataObjectImporter $data_importer
+   *   The data importer.
+   * @param \Drupal\media_mpx\DataObjectFactoryCreator $dataObjectFactoryCreator
+   *   Data factory creator.
+   */
+  public function __construct(array $configuration, string $plugin_id, $plugin_definition, EntityTypeManagerInterface $entityTypeManager, DataObjectImporter $data_importer, DataObjectFactoryCreator $dataObjectFactoryCreator) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->entityTypeManager = $entityTypeManager;
```

Comment:
A small thing, while Drupal is OK with either camel case or underscores, it's a good practice not to mix and match them in the same function or class.